packages/rps/src/redux/game/saga.ts

```diff
@@ -34,7 +34,10 @@ const isPlayersTurnNext = (
 };
 
 export function* gameSaga(client: RPSChannelClient) {
-  yield takeEvery('*', gameSagaRun, client);
+  while (true) {
+    yield take('*');
+    yield* gameSagaRun(client);
+  }
```

Comment:
yeah. It turned out this was important, as otherwise the saga would run in an unexpected mid-calculation state. I'm worried that this might mean it's a bit fragile, in particular it's possible that the saga might miss an event that happened to arrive while it was being processed. Probably the 'right' solution here is to rewrite the saga, so that it can run in mid-calculation states and still do the right thing. But this will do for now..